### DISCUSSION TITLE
ResultBeam - integration parameters

### DISCUSSION THREAD

**Discussion by jlemeriel** [Dlubal-Software/RFEM_Python_Client#365] [Q&A] [answered]
Hello, 
I try to configure a result beam, using the member.py 
The way i want to configure is "MemberResultBeamIntegration.INTEGRATE_FROM_LISTED_OBJECT". A 'list is expected but which format ? 
There is a lot of option.  (surfaces, member, solud:  included, exclude) 
Frem the memberpy : integration_parameters: list = None

**Comment by heetrojivadiya** [comment, marked as answer]
Hii @jlemeriel, 

the `integration_parameters` is used for other three types of result beam integration. But you can define include or exclude objects (surfaces, solids, members) with help of ``params`` as dictionary. You may check required keys of dictionary for those parameters in below sample code:

```py
from RFEM.initModel import Model, ConvertToDlString
from RFEM.enums import MemberResultBeamIntegration
from RFEM.BasicObjects.member import Member

Model(False, 'Demo1')

parameters = {'result_beam_include_all_surfaces': True,
              'result_beam_include_all_solids': True,
              'result_beam_include_all_members': False,                 # for custom members set False
              'result_beam_include_surfaces': ConvertToDlString(''),
              'result_beam_include_solids': ConvertToDlString(''),
              'result_beam_include_members': ConvertToDlString('2'),    # member number as str eg. '1 2 3'
              'result_beam_exclude_surfaces': ConvertToDlString(''),
              'result_beam_exclude_solids': ConvertToDlString(''),
              'result_beam_exclude_members': ConvertToDlString('1')}    # member number as str eg. '1 2 3'

Member.ResultBeam(3, 3,4, result_beam_integrate_stresses_and_forces=MemberResultBeamIntegration.INTEGRATE_FROM_LISTED_OBJECT, params=parameters)

```
![image](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/108512789/23414d32-b175-4c56-8307-6ed0a3eb3634)

Let us know if it works or not.

**Reply by jlemeriel** [reply]
Indeed, it seems to work! Except for the IncludeObjects. I can't configure any object in there

**Reply by heetrojivadiya** [reply]
For better investigation, may you please share code or snapshot of issue where you face problem?

**Reply by jlemeriel** [reply]
![2024-05-14 10_47_51-Window](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/167471486/dda84a95-7182-47c3-93f4-6ed4e6119254)
![2024-05-14 10_47_18-Spyder (Python 3 11)](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/167471486/2b9918ba-5d94-4e26-a150-0b8945f17fd2)

**Reply by heetrojivadiya** [reply]
Everything seems fine in code. It's strange behavior of it. I have tried with multiple members for include objects and it also worked for me. It should work for you too. 
![image](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/108512789/355d57bf-63f5-4483-8a87-68036f84b2b7)


Which RFEM version are you using? and may you please share code with model so I can investigate in detail?

**Reply by jlemeriel** [reply]
I'm running 6.05.0004. I'll install the last one

**Reply by heetrojivadiya** [reply]
Okay, I am using 6.06.0006. So, please try in this or latest version and let me know if it works or not.

**Reply by jlemeriel** [reply]
It works with the last version. Thanks for your support!
